### PR TITLE
Add Users management to provider 

### PR DESCRIPTION
**Pull request by Pierrelefort**
# Description

Related to #1 

- [x] Data source users
- [x] Resource user
- [x] package clients to communicate with thebastion

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit tests
- [x] Acceptance tests

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have written tests for the code I added or updated.
- [x] I have updated the documentation according to my changes.
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: thebastion/tests/utils_testing.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	}`, resourceName, uid, name, strings.Join(ingress_keys, "\",\""))
+}
+
+func TestAccTheBastionUserDataSource(exampleResource string) string {
+	return exampleResource + `
+	data "thebastion_users" "all" {}
+	`
+}
+
+// testAccPreCheck validates the necessary test API keys exist
```
Comment (simonvadee) [inline thebastion/tests/utils_testing.go]:
```suggestion
// TestAccPreCheck validates the necessary test API keys exist
```

File: thebastion/users/users_data_source_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	"terraform-provider-thebastion/thebastion/tests"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+// example.User represents a concrete Go type that represents an API resource
```
Comment (simonvadee) [inline thebastion/users/users_data_source_test.go]:
```suggestion
// TestAccTheBastionUsers_single does...example.User represents a concrete Go type that represents an API resource
```

File: utils/utils_misc_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		"zero elements": {
+			input:    make([]interface{}, 0),
+			expected: []string{},
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
```
Comment (simonvadee) [inline utils/utils_misc_test.go:85]: useless

File: utils/utils_misc_test.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+			input_right:         []string{"2", "3", "4"},
+			expected_left_only:  []string{"0", "1"},
+			expected_right_only: []string{"3", "4"},
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
```
Comment (simonvadee) [inline utils/utils_misc_test.go:134]: remove

File: utils/utils_misc_test.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+			got_left, got_right := utils.CompareLists(test.input_left, test.input_right)
+
+			if diff := cmp.Diff(got_left, test.expected_left_only); diff != "" {
+				t.Errorf("unexpected diff (+wanted, -got): %s", diff)
+			}
```
Comment (simonvadee) [inline utils/utils_misc_test.go]: I think testify has a way of doing this

File: utils/utils_misc_test.go (hunk 4 above)
Comment (Pierrelefort) [reply utils/utils_misc_test.go:134]: It does have an impact. ```Parallel signals that this test is to be run in parallel with (and only with) other parallel tests. ``` So in this case, each sub function will be paralleled when called.

**Pull request merged**